tests/MongoDB.Driver.Core.Tests/Core/Compression/CompressorsTests.cs

```diff
@@ -30,6 +32,91 @@ public class CompressorsTests
     {
         private static string __testMessage = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz";
 
+        [Theory]
+        [InlineData(CompressorType.Snappy)]
+        public void Compressor_should_read_the_previously_written_big_message_or_throw_the_exception_if_the_current_platform_is_not_supported(CompressorType compressorType)
+        {
+            var message = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz";
+
+            var compressedMessage = "158;2;100;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;254;26;0;254;26;0;254;26;0;254;26;0;1;26";
+            var decompressedMessage = "97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122";
+
+            string TestCase(string incomeMessage, string expectedCompressed, string expectedDecompressed)
+            {
+                var bytes = Encoding.ASCII.GetBytes(incomeMessage);
+                var compressor = GetCompressor(compressorType);
+
+                var inputStream = GetStreamFromBytes(bytes);
+                var outpustStream = GetStreamFromBytes();
+
+                compressor.Compress(inputStream, outpustStream);
+
+                var compressed = GetBytesFromStream(outpustStream);
```

Comment:
`compressed` contains bytes without `snappy` headers which were skipped in this line: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/55/files#diff-d915095ced45bd3f5c4fdc401621a8b3R40